### PR TITLE
Document how to authenticate with the REST api

### PR DESCRIPTION
**Pull request by Duckle29**

### CODE REVIEW

File: docs/source/rest.rst [hunk 1]
```diff
@@ -18,6 +18,9 @@ Currently the REST API provides the following API Endpoints::
     POST /apis/rest/items/<itemname>/unlock
 
 
+Authentication
+==============
+For endpoints that require non-default permissions authentication can be done through the http basic authentication, with the user ``user`` and the password for the intended permissions.
```
Comment (ThomasWaldmann) [inline docs/source/rest.rst:23]: Thanks for adding this to the docs (even I forgot about how that would work). But it needs a little fix:

From the 0.3.0 changelog entry:

support http basic auth header (it just reads the password from there, **the user name is ignored**).

this is useful for scripting, e.g. you can do now: $ curl -F 'file=@somefile;type=text/plain' http://user:password@localhost:5000/+upload

Comment (Duckle29) [reply docs/source/rest.rst]: Cool I made the change, do you want the example with curl to be added as well?

Comment (ThomasWaldmann) [reply docs/source/rest.rst]: No, people can look that up in the docs of the tool they use.

**Comment by ThomasWaldmann** [comment]
Thanks!

**Pull request merged**